### PR TITLE
Nested errors representation and migration to thiserror

### LINKED ISSUE #28: Nested Validation Error Output
I have Actix-Web web services properly responding with HTTP 400 codes and descriptive error message when validation fails for simple use cases.  However, when using nested validation (struct with derived Validate containing another struct with derived Validate), the error message returned by Actix-Web-Validator is not  very descriptive.

Please consider the following stucts:
```rust
#[derive(Serialize, Deserialize, Validate, Debug)]
#[serde(rename_all = "camelCase")]
pub struct SearchParams {
    #[validate]
    page_params: PageParams,
    determine_total_count: bool
}

#[derive(Serialize, Deserialize, Validate, Debug)]
#[serde(rename_all = "camelCase")]
pub struct PageParams {
    #[validate(range(min = 0))]
    page: u16,
    #[validate(range(min = 1, max = 100))]
    page_size: u8
}
```

If I have a web service with input param Json\<PageParams\> and validation fails, I get a descriptive error message such as this:
>Validation errors in fields:
>	pageSize: range

However, if I have a web service with input param Json\<SearchParams\> and validation fails on the same page_size field of the PageParams struct (nested validation), then the error message does not indicate which nested field failed validation or why:
>Validation errors in fields:

### PR DESCRIPTION
**Pull request by singulared**
Close #28

### CODE REVIEW

File: src/error.rs [hunk 1]
```diff
@@ -33,51 +34,67 @@ impl From<serde_json::error::Error> for Error {
...
+    _flatten_errors(errors, None)
 }
 
-impl ResponseError for Error {
-    fn error_response(&self) -> HttpResponse {
-        HttpResponse::build(StatusCode::BAD_REQUEST).body(match self {
-            Self::Validate(e) => format!(
-                "Validation errors in fields:\n{}",
-                e.field_errors()
+fn _flatten_errors(
```
Comment (Roms1383) [inline src/error.rs:68]: you probably want to `#[inline]` this method.

File: src/error.rs [hunk 2]
```diff
@@ -33,51 +34,67 @@ impl From<serde_json::error::Error> for Error {
...
 }
 
-impl From<actix_web::error::UrlencodedError> for Error {
-    fn from(error: actix_web::error::UrlencodedError) -> Self {
-        Error::UrlEncodedError(error)
-    }
+/// Helper function for error extraction and formatting.
+/// Return Vec of tuples where first element is full field path (separated by dot)
+/// and second is error.
+pub fn flatten_errors(errors: &ValidationErrors) -> Vec<(String, &ValidationError)> {
```
Comment (Roms1383) [inline src/error.rs:64]: you probably want to `#[inline]` this method.

**Review by Roms1383 — COMMENTED** [review]
As stated on #35 this is indeed way better than relying on indentation, kudos.
I'd probably just add inlining annotations (see my comments) to avoid paying the cost of functions calls.
As of the dependency introduced if all maintainers are ok it should be fine (I have no authority here).

**Comment by singulared** [comment]
Anyway, I added indentation too into `flatten_error` function output.

**Pull request merged**